web3/middleware/filter.py

```diff
@@ -59,8 +59,8 @@ def gen_bounded_segments(start, stop, step):
         yield (start, stop)
         return
     for segment in zip(
-            range(start, stop - step, step),
-            range(start + step, stop, step)):
+            range(start, stop - step + 1, step),
+            range(start + step, stop + 1, step)):
         yield segment
     else:
```

Comment:
Its a for/else loop, so it runs at the end of the for loop.  It would work the same to remove the else clause, but probably safer to keep it in case anything is added to this method.

Reply:
`for/else` is typically reserved for loops that have a `break` somewhere in them, in order to *only* run the `else` in whatever case the `break` does not get triggered (or inversely, having the ability to `break`, thus skipping the `else` code.